apisix/core/config_etcd.lua

```diff
@@ -463,6 +480,76 @@ local function sync_data(self)
         end
 
         self.conf_version = self.conf_version + 1
+
+        if self.key ~= "/apisix/routes" then
+            goto CONTINUE
+        end
+
+        local route
+        if res.value then
+            local status = table.try_read_attr(res, "value", "status")
+            if status and status == 0 then
+                goto CONTINUE
+            end
+
+            local filter_fun, err
+            if res.value.filter_func then
+                filter_fun, err = loadstring(
+                    "return " .. res.value.filter_func,
+                    "router#" .. res.value.id
+                )
+                if not filter_fun then
+                    log.error("failed to load filter function: ", err, " route id", res.value.id)
+                        goto CONTINUE
+                end
+
+                filter_fun = filter_fun()
+            end
+
+            route = {
+                id = res.value.id,
+                paths = res.value.uris or res.value.uri,
+                methods = res.value.methods,
+                priority = res.value.priority,
+                hosts = res.value.hosts or res.value.host,
+                remote_addrs = res.value.remote_addrs or res.value.remote_addr,
+                vars = res.value.vars,
+                filter_fun = filter_fun,
+                handler = function(api_ctx, match_opts)
+                    api_ctx.matched_params = nil
+                    api_ctx.matched_route = res
+                    api_ctx.curr_req_matched = match_opts.matched
+                end
+            }
+        end
+
+        local router_opts = {
+            no_param_match = true
+        }
+        local router_mod = require("apisix.router")
+        if op == 2 then
+            log.notice("create routes watched from etcd into radixtree.", json.encode(res))
+            err = router_mod.uri_router:add_route(route, router_opts)
```

Comment:
Where's the source code of the new methods of the router, e.g. `add_route`?

Reply:
add_route and other methods to operate radixtree is in this path apisix/deps/share/lua/5.1/resty/radixtree.lua .
the source code is here.  https://pan.baidu.com/s/1L2dxC67PoaJ3brsF-sDDXQ    enter code: 6688

Reply:
Please refer to my suggestion below about code refacotring, thanks.

Reply:
I've implemented the method to delete the node in radixtree. The share library is in apisix/deps/lib/lua/5.1/librestyradixtree.so.  It can be download in https://pan.baidu.com/s/1F0344pwPFx3pzfbNpv6Nzw    enter code: 5566